src/js/Framework/PasswordStrengthChecker.js

```diff
@@ -0,0 +1,49 @@
+import axios from 'axios'
+import { debounce } from 'lodash'
+
+export default function passwordStrengthChecker () {
+  const passwordInput = document.querySelectorAll('[data-role="check-password"] input[type="password"]')[0]
+  const meterSections = document.querySelectorAll('.meter-section')
+
+  const debounced = debounce(getStrength, 250, { maxWait: 750, leading: true, trailing: true })
+  if (passwordInput) passwordInput.addEventListener('input', debounced)
+
+  function getStrength () {
+    const password = passwordInput.value
+    const dataRoute = passwordInput.closest('[data-route]').dataset.route
+
+    axios.post(dataRoute, {
+      password: password
+    })
+      .then(function (response) {
+        updateMeter(response.data.strength)
+      })
+      .catch(function (error) {
+        console.error(error)
+      })
+  }
+
+  function updateMeter (strength) {
+    // Remove all strength classes
+    meterSections.forEach((section) => {
+      section.classList.remove('weak', 'medium', 'strong', 'very-strong')
+    })
+
+    // Add the appropriate strength class based on the strength value
+    if (strength === 0) {
+      meterSections[0].classList.add('weak')
```

Comment:
is onderstaande niet logischer

```javascript
const classes = [ 'weak', 'medium', 'strong', 'very-strong')
for (let i =; i < strength; i++) {
  meterSections[I].classList.add(classes[strength])
}
```

en dan kan je die `classes`, vermoedelijk ook gebruiken bij het verwijderen van de classes. op lijn 28.

Reply:
Jep is wel compacter, ik had het overwogen maar vond het leesbaarder in het lang. Ik vervang het.

Reply:
Doe gerust wat je zelf logisch vind, uwe is inderdaad leesbaarder. Denk dat mijn oplossing compacter is en "uitbreidbaarder" als er ooit een extra class komt.